src/main/resources/db/changesets/20220411_requiredcolumns.yaml

```diff
@@ -0,0 +1,21 @@
+databaseChangeLog:
+  - changeSet:
+      id: self_hosted_datasets
+      author: tlangs
+      changes:
+        - addColumn:
+            tableName: dataset_column
+            columns:
+              name: required
+              type: boolean
+              defaultValue: false
+              constraints:
+                nullable: false
+        - addColumn:
```

Comment:
Actually, just realized something: we presumably need to make the migration a little smarter and convert primary key columns to be required=true.

We also probably need a validator to ensure that we don't allow creating non-require PK field

Reply:
For your second point, I think it makes sense to just make PK fields required, and ignore the `required` value set by the user. We have no clue if a `false` value for `required` on a PK field is the default, or if the user actually set it themselves.

Reply:
> We have no clue if a false value for required on a PK field is the default, or if the user actually set it themselves.

That's a good point 🤔 We should mention PK fields default to true in the swagger description